_posts/2019-07-17-#15169.md

```diff
@@ -7,7 +7,29 @@ title: "#15169 Parallelize CheckInputs() in AcceptToMemoryPool()"
 
 ## Notes
 
+* The [CheckInputs()](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1234) function is called when checking the inputs to a transaction. Each input has a script which is verified. That verification can be run:
+    - sequentially and synchronously, in which case `CheckInputs()` will return `true` if all scripts pass verification or pass back the error for the first script that failed verification; or
+    - in parallel, in which case `CheckInputs()` will emplace the script check onto a queue for future verification.
+* Previously, `CheckInputs()` would always check inputs sequentially when accepting a transaction to the mempool. Transaction inputs are checked in parallel during block validation. Note the [`vChecks` argument when calling `CheckInputs()` from `ConnectBlock()`](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1887).
+* This PR enables the input checks to be performed in parallel when a transaction is entering the mempool. 
+* It does so by replacing calls to `CheckInputs()` with [`RunCheckInputsMaybeParallel()`](https://github.com/bitcoin/bitcoin/commit/cb0c42cfda7669f3df57c167cf0d691f926039d2#diff-24efdb00bfbe56b140fb006b562cc70bR1679), which will push the input checks to an existing CCheckQueue worker queue.
+  * [`CheckInputs`](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1234)
+  * [`CCheckQueue`](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/validation.cpp#L1528) (Currently used for Block validation)
+  * [Script check worker loop](https://github.com/bitcoin/bitcoin/blob/8f9725c83f1da5364354d2820696c84d542d3d37/src/checkqueue.h#L66) (Currently used for Block validation)
+* There is a significant performance gain resulting from this optimization.
+* This PR also changes behavior when transactions are denied mempool acceptance. `RunCheckInputsMaybeParallel` (and by extension [`CheckInputs()`](https://github.com/bitcoin/bitcoin/pull/15169/commits/5f4e514412fc39e174f298a7737eb9f08a82a86b#diff-24efdb00bfbe56b140fb006b562cc70bL1436)) no longer set TX_NOT_STANDARD during input check failure, but only a consensus failure state.
+  * Note: This has an effect on peer connections. Peers are not dropped by a node when it receives a non-standard but consensus-valid transaction. Instead, a peer will only be dropped when consensus-invalid transactions are received. Since input checks for new transactions
```

Comment:
Is there text missing after "Since input checks for new transactions"...?

Reply:
Thanks again. Not my finest hour.